src/models/VisualStyleModel/impl/VisualStyleImpl.ts

```diff
@@ -469,62 +277,627 @@ export const createVisualStyleFromCx = (cx: Cx2): VisualStyle => {
+): {
+  defaultStyle: Stylesheet[]
+  cyNodes: ElementDefinition[]
+  cyEdges: ElementDefinition[]
+  nodeBypasses: Record<IdType, Partial<Record<string, VisualPropertyValueType>>>
+  edgeBypasses: Record<IdType, Partial<Record<string, VisualPropertyValueType>>>
+} => {
+  // edge ids are of the form 'e1', 'e2', etc. but our app stores them as '1', '2', etc.
+  const nodeStyle: Record<
+    string,
+    VisualPropertyValueType | ((node: NodeSingular) => VisualPropertyValueType)
+  > = {}
+  const edgeStyle: Record<
+    string,
+    VisualPropertyValueType | ((edge: EdgeSingular) => VisualPropertyValueType)
+  > = {}
+
+  const nodeBypasses: Record<
+    IdType,
+    Partial<Record<string, VisualPropertyValueType>>
+  > = {}
+  const edgeBypasses: Record<
+    IdType,
+    Partial<Record<string, VisualPropertyValueType>>
+  > = {}
+
+  nodeVisualProperties(vs).forEach((vpName: VisualPropertyName) => {
+    const vp = vs[vpName] as VisualProperty<VisualPropertyValueType>
+    const defaultValue = vp.defaultValue
+    const mapping = vp.mapping
+    const bypassMap = vp.bypassMap
+    const cyStyleName = cyJsVisualPropertyConverter[vpName]?.cyJsVPName
+
+    if (cyStyleName != null) {
+      nodeStyle[cyStyleName] = defaultValue
+
+      if (mapping != null) {
+        switch (mapping.type) {
+          case 'passthrough': {
+            const { attribute } = mapping
+            nodeStyle[cyStyleName] = (
+              node: NodeSingular,
+            ): VisualPropertyValueType => {
+              const row = nodeTable.rows.get(node.data('id'))
+              const column = nodeTable.columns.get(attribute)
+              const value =
+                (column?.alias != null
+                  ? row?.[column.alias]
+                  : row?.[attribute]) ?? column?.defaultValue
+
+              if (!Array.isArray(value) && value != null) {
+                return value
+              }
+              throw new Error(
+                `Error applying passthrough mapping function on node ${
+                  node.data('id') as string
+                }.  Value: ${String(
+                  value,
+                )} should not be an array or undefined`,
+              )
+            }
+            break
+          }
+          case 'discrete': {
+            const { attribute } = mapping
+            const m = mapping as DiscreteMappingFunction
+            nodeStyle[cyStyleName] = (
+              node: NodeSingular,
+            ): VisualPropertyValueType => {
+              const row = nodeTable.rows.get(node.data('id'))
+              const column = nodeTable.columns.get(attribute)
+              const value =
+                (column?.alias != null
+                  ? row?.[column.alias]
+                  : row?.[attribute]) ?? column?.defaultValue
+
+              if (!Array.isArray(value) && value != null) {
+                return m.vpValueMap.get(value) ?? defaultValue
+              }
+
+              throw new Error(
```

Comment:
Undefined values should not throw an error.  Instead return default style values.